TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php

```diff
@@ -115,6 +115,10 @@ public function getContextPath() {
 	 * @api
 	 */
 	public function setName($newName) {
+		if ($this->getName() === $newName) {
+			return;
+		}
+
```

Comment:
Yes, it was moved further up, see full diff. It makes sense to check that first before anything else.